modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/requestPathAuthenticator/RequestPathAuthenticatorTestCase.java

```diff
@@ -0,0 +1,308 @@
+        attributeConsumerServiceIndexProp.setValue("1239245949");
+        requestConfig.setProperties(new Property[]{attributeConsumerServiceIndexProp});
+
+        inboundAuthenticationConfig.setInboundAuthenticationRequestConfigs(
+                new InboundAuthenticationRequestConfig[]{requestConfig});
+
+        serviceProvider.setInboundAuthenticationConfig(inboundAuthenticationConfig);
+        RequestPathAuthenticatorConfig requestPathAuthenticatorConfig = new RequestPathAuthenticatorConfig();
+        requestPathAuthenticatorConfig.setName("BasicAuthRequestPathAuthenticator");
+
+        serviceProvider.setRequestPathAuthenticatorConfigs(new
+                RequestPathAuthenticatorConfig[]{requestPathAuthenticatorConfig});
+        appMgtclient.updateApplicationData(serviceProvider);
+        serviceProvider = appMgtclient.getApplication(SERVICE_PROVIDER_NAME);
+
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void atEnd() throws Exception {
+        appMgtclient.deleteApplication(serviceProvider.getApplicationName());
+        if (tomcat != null) {
+            tomcat.stop();
+            tomcat.destroy();
+            Thread.sleep(10000);
+        }
+    }
+
+    @Test(alwaysRun = true, description = "Test login success")
+    public void testLoginSuccess() throws Exception {
+        HttpPost request = new HttpPost(TRAVELOCITY_SAMPLE_APP_URL + "/samlsso?SAML2.HTTPBinding=HTTP-POST");
+        List<NameValuePair> urlParameters = new ArrayList<>();
+        urlParameters.add(new BasicNameValuePair("username", adminUsername));
+        urlParameters.add(new BasicNameValuePair("password", adminPassword));
+        request.setEntity(new UrlEncodedFormEntity(urlParameters));
+        HttpResponse response = client.execute(request);
+        BufferedReader rd = new BufferedReader(new InputStreamReader(response.getEntity().getContent()));
+        String line;
+        String samlRequest = "";
+        String secToken = "";
+
+        while ((line = rd.readLine()) != null) {
+            if (line.contains("name='SAMLRequest'")) {
+                String[] tokens = line.split("'");
+                samlRequest = tokens[5];
+            }
+            if (line.contains("name='sectoken'")) {
+                String[] tokens = line.split("'");
+                secToken = tokens[5];
+            }
+        }
+        EntityUtils.consume(response.getEntity());
+        request = new HttpPost(isURL + "samlsso");
+        urlParameters = new ArrayList<>();
+        urlParameters.add(new BasicNameValuePair("sectoken", secToken));
+        urlParameters.add(new BasicNameValuePair("SAMLRequest", samlRequest));
+        request.setEntity(new UrlEncodedFormEntity(urlParameters));
+        response = client.execute(request);
+        int responseCode = response.getStatusLine().getStatusCode();
+        Assert.assertEquals(responseCode, 200, "Successful login response returned code " + responseCode);
```

Comment:
Specifically check for the status code to be success in the SAML Response. 200ok, would be given even in a failure response.